frontend/src/layouts/dark/main/DarkMainLayout.tsx

```diff
@@ -3,18 +3,89 @@
 import { DARK_MAIN_LAYOUT_CLASS_NAME } from "../../../constants/appClassName";
 import { LayoutProps } from "../../../types/DashboardLayout";
 import { AppHeader } from "../../../components/common/layout/appHeader/AppHeader";
+import {LogService} from "../../../fi/hg/core/LogService";
+import {useParams} from "react-router-dom";
+import {useUrlWorkspaceName} from "../../../hooks/workspace/useUrlWorkspacename";
+import {useEventCurrentWorkspaceChanged} from "../../../hooks/workspace/useEventCurrentWorkspaceChanged";
+import {useCallback, useState} from "react";
+import {RouteService} from "../../../fi/hg/frontend/services/RouteService";
+import {useCurrentWorkspaceName} from "../../../hooks/workspace/useCurrentWorkspaceName";
+import {AppModalType} from "../../../types/AppModalType";
+import {MY_WORKSPACE_LIST_ROUTE} from "../../../constants/route";
+import {AppModalService} from "../../../services/AppModalService";
 import "./DarkMainLayout.scss";
 
+const LOG = LogService.createLogger('DarkMainLayout');
 export function DarkMainLayout (props: LayoutProps) {
     const children = props.children;
     const t = props.t;
+
+    // urls
+    const workspaceName = useCurrentWorkspaceName();
+
+    const opts = useParams<string>();
+    const userId = opts?.id ?? undefined;
+    const workspaceId = opts?.parentId ?? undefined;
+
+    const [user, setUserId] = useState<string | undefined>(userId);
+    const [name, setName ] = useState<string | undefined>(workspaceName);
+    const [urlWorkspace]  = useUrlWorkspaceName(workspaceId);
+
+    const workspaceCallback = useCallback(
+        () => {
+            LOG.info(`Workspace changed`);
+            if (urlWorkspace !== undefined || workspaceId !== undefined) {
+                setName(urlWorkspace?.name ?? workspaceName);
+            }
+
+            if (urlWorkspace !== undefined ) {
+                LOG.info(`Workspaces changed; Moving to workspace view`, urlWorkspace);
+                setName(urlWorkspace?.name ?? workspaceName);
+
+                // if url with user
+                if(user !== undefined ){
+                    LOG.info("Open modal with userId", user);
+
+                    RouteService.setRoute('/workspace/'+ urlWorkspace.id +'/users/'+ user);
```

Comment:
Use constant functions from `constants/route` for these routes

Reply:
Also not sure if the view route should even change in this case. At least it should not move to the user view but user list view. We don't even have a user view yet. Anyway this code seems to be in wrong place since this is a DarkLayout component and it should not have this type of logic at all.